expression/util.go

```diff
@@ -587,3 +587,20 @@ func GetIntFromConstant(ctx sessionctx.Context, value Expression) (int, bool, er
 	}
 	return intNum, false, nil
 }
+
+func CoveredBySchema(expr Expression, schema *Schema) bool {
+	switch v := expr.(type) {
+	case *Column:
+		return schema.ColumnIndex(v) != -1
+	case *CorrelatedColumn:
```

Comment:
I think we don't need to check correlated column

Reply:
for completeness, I think we should.